pkg/execution/execution.go

```diff
@@ -264,3 +264,32 @@ func (h HandlePauseResult) Processed() int32 {
 func (h HandlePauseResult) Handled() int32 {
 	return h[1]
 }
+
+type GracefulError struct {
```

Comment:
I would have liked the `expressions.CompileError` to implement the GracefulError, however, this would require importing packages or code that isn't immediately related, so we have to convert the "internal" errors to graceful/public/user-facing ones one layer above.

Happy to accept suggestions on whether we should refactor this into a separate module and start implementing a common interface.